compiler/rustc_trait_selection/src/traits/query/dropck_outlives.rs

```diff
@@ -173,11 +192,25 @@ pub fn compute_dropck_outlives_inner<'tcx>(
         // do not themselves define a destructor", more or less. We have
         // to push them onto the stack to be expanded.
         for ty in constraints.dtorck_types.drain(..) {
-            let Normalized { value: ty, obligations } =
-                ocx.infcx.at(&cause, param_env).query_normalize(ty)?;
-            ocx.register_obligations(obligations);
+            let ty = if report_errors {
+                let normalized_ty = ocx.deeply_normalize(&cause, param_env, ty)?;
 
-            debug!("dropck_outlives: ty from dtorck_types = {:?}", ty);
+                let errors = ocx.select_where_possible();
+                if !errors.is_empty() {
+                    debug!("failed to normalize dtorck type: {ty} ~> {errors:#?}");
+                    return Err(errors);
+                }
+                normalized_ty
+            } else if let Ok(Normalized { value: ty, obligations }) =
```

Comment:
Could you invert this? First use query normalization, otherwise `if report_errors` recompute the errors from deeply normalizing, or otherwise return an empty vec?

Also, does it really matter if we recompute the errors unconditionally? Presumably we're never computing dropck outlives and expecting it to fail even if the program passes, right?